src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -41,4 +44,44 @@ public void shooterOut(double shooterSpeed) {
             shooterMotors.set(shooterSpeed);
         }
     }
+
+        // increments baseSpeed of shooter motors in 4 zones of left trigger axis
+    public static void shooterVelocity (double shooterVelocity) {
+        double incrementedSpeed = 0;
+        if (shooterVelocity < 0.25 && shooterVelocity > 0) {
+            incrementedSpeed = Constants.ShooterConstants.lowestSpeed;
+        } else if (shooterVelocity < 0.5 && shooterVelocity > 0) {
+            incrementedSpeed = (Constants.ShooterConstants.lowestSpeed + Constants.ShooterConstants.increment);
+        } else if (shooterVelocity < 0.75 && shooterVelocity > 0) {
+            incrementedSpeed = (Constants.ShooterConstants.lowestSpeed + Constants.ShooterConstants.increment * 2);
+        } else if (shooterVelocity <= 1 && shooterVelocity > 0) {
+            incrementedSpeed = (Constants.ShooterConstants.lowestSpeed + Constants.ShooterConstants.increment * 3);
+        } else {
+            incrementedSpeed = 0;
+        }
+
+        if (incrementedSpeed > baseSpeed) {
+            baseSpeed = incrementedSpeed;
+            updateMotors();
+        }
```

Comment:
Here we should add `updateMotors();` to make sure the motors are always receiving a value.